sphinxlint.py

```diff
@@ -196,124 +197,206 @@ def is_in_a_table(error, line):
+    """Search for hyperlinks missing underscore after their closing backtick."""
+    for lno, line in enumerate(lines, start=1):
+        if "`" not in line:
+            continue
+        for match in seems_hyperlink_re.finditer(line):
             if not match.group(2):
                 yield lno, "missing underscore after closing backtick in hyperlink"
+
+
+@checker(".rst")
+def check_roles_with_double_backtick(file, lines):
+    """Search for roles with double backticks like: :fct:``sum``."""
+    for lno, line in enumerate(lines, start=1):
+        if "`" not in line:
+            continue
         if double_backtick_role.search(line):
             yield lno, "role use a single backtick, double backtick found."
+
+
+@checker(".rst")
+def check_roles_missing_leading_space(file, lines):
+    """Search for spaces missing before roles, like the:fct:`sum`."""
+    for lno, line in enumerate(lines, start=1):
+        if "`" not in line:
+            continue
         if role_glued_with_word.search(line):
             yield lno, "missing space before role"
-        if role_missing_right_column.search(line):
-            yield lno, "role missing column before first backtick."
+
+
+@checker(".rst")
+def check_roles_missing_colon_before_backtick(file, lines):
+    """Search for roles missing a colon, like: :issue`123`."""
+    for lno, line in enumerate(lines, start=1):
+        if role_missing_right_colon.search(line):
+            yield lno, "role missing colon before first backtick."
 
 
 @checker(".py", ".rst", rst_only=False)
-def check_whitespace(file, lines):
-    """Check for whitespace and line length issues."""
-    lno = line = None
+def check_carriage_returns(file, lines):
+    """Check for carriage return in lines."""
     for lno, line in enumerate(lines):
         if "\r" in line:
             yield lno + 1, "\\r in line"
+
+
+@checker(".py", ".rst", rst_only=False)
+def check_horizontal_tabs(file, lines):
+    """Check for horizontal tabs in lines."""
+    for lno, line in enumerate(lines):
         if "\t" in line:
             yield lno + 1, "OMG TABS!!!1"
+
+
+@checker(".py", ".rst", rst_only=False)
+def check_trailing_whitespaces(file, lines):
+    """Check for trailing whitespaces at end of lines."""
+    for lno, line in enumerate(lines):
         if line.rstrip("\n").rstrip(" \t") != line.rstrip("\n"):
             yield lno + 1, "trailing whitespace"
-    if line is not None:
-        if not line.endswith("\n"):
-            yield lno, "No newline at end of file (no-newline-at-end-of-file)."
 
 
-@checker(".rst", severity=0, rst_only=False)
-def check_line_length(file, lines):
+@checker(".py", ".rst", rst_only=False)
+def check_newline_at_end_of_file(file, lines):
+    """Check if the last line ends with a newline, like any other lines."""
+    if lines:
+        if not lines[-1].endswith("\n"):
+            yield len(lines), "No newline at end of file (no-newline-at-end-of-file)."
```

Comment:
Other messages don't include the `(no-newline-at-end-of-file)` part, and this doesn't match the checker name.

Reply:
:+1: